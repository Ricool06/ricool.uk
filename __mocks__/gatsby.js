import React from 'react';

const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(queryString => queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) => React.createElement('a', {
    ...rest,
    href: to,
  })),
  StaticQuery: jest.fn(),
};
