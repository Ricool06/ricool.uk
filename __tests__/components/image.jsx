import React from 'react';

import { StaticQuery } from 'gatsby';
import renderer from 'react-test-renderer';
import { render, cleanup } from 'react-testing-library';
import Image, { PureImage } from '../../src/components/image';

describe('Image', () => {
  const filePath = 'liontraceroar.png';
  const data = {
    image: {
      childImageSharp: {
        fluid: {
          src: filePath,
          aspectRatio: 1,
          srcSet: filePath,
          sizes: '(max-width: 300px) 100vw, 300px',
        },
      },
    },
  };

  afterEach(cleanup);

  it('should render the image at the path specified', () => {
    const { container } = render(<PureImage data={data} />);
    expect(container.querySelector('img')).toHaveAttribute('src', filePath);
  });

  it('should match the snapshot', () => {
    StaticQuery.mockImplementation(({ render: mockRender }) => mockRender(data));

    const jsonSnapshot = renderer.create(<Image filePath={filePath} />).toJSON();
    expect(jsonSnapshot).toMatchSnapshot();
  });
});
