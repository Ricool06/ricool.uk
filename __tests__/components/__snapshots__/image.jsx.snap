// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match the snapshot 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  <picture>
    <source
      sizes="(max-width: 300px) 100vw, 300px"
      srcSet="liontraceroar.png"
    />
    <img
      alt=""
      onLoad={[Function]}
      src="liontraceroar.png"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 0.5s",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcSet=\\"liontraceroar.png\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img src=\\"liontraceroar.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
